cmd/server_test.go

```diff
@@ -1680,10 +1680,42 @@ func (s *TestSuiteCommon) TestListObjectsHandler(c *check) {
 
 	getContent, err := ioutil.ReadAll(response.Body)
 	c.Assert(err, nil)
+	c.Assert(strings.Contains(string(getContent), "<Prefix></Prefix>"), true)
 	c.Assert(strings.Contains(string(getContent), "<Key>bar</Key>"), true)
 
+	// create listObjectsV1 request with prefix "/"
+	request, err = newTestSignedRequest("GET", getListObjectsV1URL(s.endPoint, bucketName, "1000", "/"),
+		0, nil, s.accessKey, s.secretKey, s.signer)
+	c.Assert(err, nil)
+	client = http.Client{Transport: s.transport}
+	// execute the HTTP request.
+	response, err = client.Do(request)
+	c.Assert(err, nil)
+	c.Assert(response.StatusCode, http.StatusOK)
+
+	getContent, err = ioutil.ReadAll(response.Body)
+	c.Assert(err, nil)
+	c.Assert(strings.Contains(string(getContent), "<Prefix>/</Prefix>"), true)
+	c.Assert(strings.Contains(string(getContent), "<Key>bar</Key>"), false)
+
+	// create listObjectsV1 request with prefix "/bar"
+	request, err = newTestSignedRequest("GET", getListObjectsV1URL(s.endPoint, bucketName, "1000", "/bar"),
+		0, nil, s.accessKey, s.secretKey, s.signer)
+	c.Assert(err, nil)
+	client = http.Client{Transport: s.transport}
+	// execute the HTTP request.
+	response, err = client.Do(request)
+	c.Assert(err, nil)
+	c.Assert(response.StatusCode, http.StatusOK)
+
+	getContent, err = ioutil.ReadAll(response.Body)
+	c.Assert(err, nil)
+	c.Assert(strings.Contains(string(getContent), "<Prefix>/bar</Prefix>"), true)
+	c.Assert(strings.Contains(string(getContent), "<Key>bar</Key>"), false)
```

Comment:
you don't need to add the tests here instead you should add them to our mint functional tests github.com/minio/mint

Reply:
You mean I should put those tests to separated SDK repertories?

Reply:
Yes we are actively these type of mock functions.